draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1062,98 +923,338 @@ trace has a different value for a given field, this field MUST NOT be added to
-AEAD tag), event definition documents SHOULD define how to support header_length
-calculation.
+Event categories MUST belong to a single event schema. The MUST have a
+registered non-empty globally-unique identifier of type `text` and MUST have a
+single dereferencable URI. That URI MUST be absolute and MUST indicate the
+category identifier using a fragment identifier (characters after a "#" in the
+URI). For event categories in schema defined by RFCs, the URI SHOULD be a URN of
+the form `urn:ietf:params:qlog:<schema namespace>#<category identifier>`.
+Private or non-standard event categories can use other URI formats. URIs that
+contain a domain name SHOULD also contain a month-date in the form mmyyyy. The
+definition of the category and assignment of the URI MUST have been authorized by
+the owner of the domain name on or very close to that date. (This avoids
+problems when domain names change ownership.)
 
-# Common events and data classes
+The registration requirements for extension schema URIs are detailed in
+{{iana}}.
 
-There are some event types and data classes that are common across protocols,
-applications, and use cases. This section specifies such common definitions.
+Concrete event types MUST belong to a single event category and MUST have a
+non-empty identifier of type `text`.
 
-~~~ cddl
-GenericEventData = GenericError /
-                GenericWarning /
-                GenericInfo /
-                GenericDebug
+The value of a qlog event `name` field MUST be the concatenation of category
+identifier, colon (':'), and event type identifier. By virtue of the identifier
+requirements, names are globally-unique. Thus, log files can contain events from
+multiple schemas without the risk of name collisions.
 
-SimulationEventData = SimulationScenario /
-                SimulationMarker
+In the following hypothetical example, a qlog file contains events belonging to
+multiple standard and non-standard categories belonging to different schema
+namespaces rick, john, and pickle. The standard categories use a URN format, the
+non-standard categories use a URI with domain name.
 
-$ProtocolEventData /= GenericEventData / SimulationEventData
 ~~~
-{: #commonevent-integration title="ProtocolEventData extension for common
-events"}
-
-## Generic events
+"event_categories": [
+                      "urn:ietf:params:qlog:rick#roll",
+                      "urn:ietf:params:qlog:rick#astley",
+                      "urn:ietf:params:qlog:rick#moranis",
+                      "urn:ietf:params:qlog:john#doe",
+                      "urn:ietf:params:qlog:john#candy",
+                      "urn:ietf:params:qlog:john#goodman",
+                      "https://example.com/032024/pickle.html#pepper",
+                      "https://example.com/032024/pickle.html#lilly",
+                      "https://example.com/032024/pickle.html#rick",
+                    ]
+~~~
+{: #event-categories title="Example event_categories serialization"}
```

Comment:
At no point has it been made clear yet how loggers MUST/SHOULD fill the `event_categories` field, nor how tools should handle "missing" URIs (i.e., unknown event types). I think this should be explicitly called out.

Reply:
Very good poont